testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -797,7 +797,7 @@
       $stderr.puts 'ERROR - try again'
       row = first(:xpath, "//div[@class=\"table-responsive\"]/table/tbody/tr[.//td[contains(.,'#{arg1}')]]")
     end
-    row.first(:xpath, './/input[@type="checkbox"]').set(true)
+    row.first(:xpath, './/input[@type="checkbox"]').set(true) unless row.nil?
```

Comment:
Well, honestly the concept behind this fix does not sound good to me.
I mean, I see the point of preventing using a `nil` element, but ignoring the fact that the row is `nil` this hides possible bugs or not expected behaviors.
I am fine to not call `set(true)` on `nil`, but whenever the row I expect to be there is `nil` this should trigger and exception anyway.

Let's look at an higher level and keep the step logic independent from the context where it is used. If I use a step like this `And I check "andromeda-dummy" in the list` :arrow_right: **here I want to `check` the element in the `list` because I really expect it to be there.** If I can't find it, it could be that the assumption of the used scenario is wrong or maybe something else has gone wrong: in anycase I want to raise the exception because not finding the row is an unexpected/undesired behavior.

Maybe for the sake of robustness this final step implementation should be:
```
// if I can find the row element I apply my action as desired
// else I raise exception because I couldn't find it
```
In the end, there are some corner case where not throwing and exception can be fine: for instance when we do some `Cleanup` where it is tolerable that some packages are not there already, etc. For this specific cases I'd implement another step that does the same logic but it does not throw the exception, and it should be shaped - for instance - like `And I check "andromeda-dummy" in the list if I can find it` where the `if I can find it` suffix does the trick to differ from the other step.

What do you think?

Reply:
Totally agree. It was a quick fix, to reduce failures in the head testsuite, but I did to fast.And I totally agree your point.